ext/php8/span.c

```diff
@@ -189,10 +188,15 @@ void ddtrace_close_span(ddtrace_span_fci *span_fci) {
         span_fci->dispatch = NULL;
     }
 
-    // A userland span might still be open so we check the span ID stack instead of the internal span stack
-    // In case we have root spans enabled, we need to always flush if we close that one (RSHUTDOWN)
-    if (DDTRACE_G(span_ids_top) == NULL && get_DD_TRACE_AUTO_FLUSH_ENABLED()) {
-        if (ddtrace_flush_tracer() == FAILURE) {
+    if (DDTRACE_G(span_ids_top) == NULL) {
+        // Enforce a sampling decision here
+        ddtrace_fetch_prioritySampling_from_root();
+
+        DDTRACE_G(root_span) = NULL;
+
+        // A userland span might still be open so we check the span ID stack instead of the internal span stack
```

Comment:
Is this still true?

Reply:
this comment is obsolete, yes.

Reply:
Actually that's untrue - we still have a couple tests relying on that behavior - we'll clean it up once we get rid of the manual span id stack.